node/reqres.js

```diff
@@ -134,7 +205,13 @@ TChannelOutgoingRequest.prototype.hookupCallback = function hookupCallback(callb
     }
     function onResponse(res) {
         self.removeListener('error', onError);
-        callback(null, res);
+        if (callback.canStream) {
```

Comment:
I dont understand the difference between streaming and non streaming here.

Why would one need zalgo protection and not the other ?